### PR TITLE
Add info.beyondallreason.bar

### PR DESCRIPTION
**Pull request by tobias47n9e**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/beyond-all-reason/BYAR-Chobby/pull/144
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*, I will still try to get the one remaining patch merged.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: info.beyondallreason.bar.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    <description>
+        <p>Beyond All Reason is an RTS</p>
+    </description>
+    <developer_name>BAR Team</developer_name>
+    <url type="homepage">https://www.beyondallreason.info</url>
+    <url type="bugtracker">https://github.com/beyond-all-reason/Beyond-All-Reason/issues</url>
+    <url type="donation">https://www.beyondallreason.info/donate-for-bar</url>
+    <screenshots>
+    </screenshots>
+    <releases>
```
Comment (hfiguiere) [inline info.beyondallreason.bar.appdata.xml:17]: it needs a release tag.

File: info.beyondallreason.bar.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+    <id>info.beyondallreason.bar</id>
+    <name>Beyond All Reason</name>
+    <summary>The Total Annihilation Inspired RTS you've been waiting for</summary>
+    <metadata_license>CC-BY-SA 4.0, CC-BY-NC-ND 4.0, and others</metadata_license>
```
Comment (hfiguiere) [inline info.beyondallreason.bar.appdata.xml:6]: this is probably not what you think. This is the license of THIS file. Prefered license for it is `CC0-1.0`

File: info.beyondallreason.bar.appdata.xml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    <project_license>GPL 2.0, GPL-3.0+</project_license>
```
Comment (hfiguiere) [inline info.beyondallreason.bar.appdata.xml:7]:
```suggestion
    <project_license>GPL 2.0 and GPL-3.0+</project_license>
```

**Comment by hfiguiere** [comment]
bot, build info.beyondallreason.bar

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: info.beyondallreason.bar.appdata.xml (hunk 2 above)
Comment (tobias47n9e) [reply info.beyondallreason.bar.appdata.xml:6]: I did mix that up with the game assets. Thanks for noticing

File: info.beyondallreason.bar.appdata.xml (hunk 1 above)
Comment (tobias47n9e) [reply info.beyondallreason.bar.appdata.xml:17]: I added one

**Comment by tobias47n9e** [comment]
Is it possible to preview the store page somewhere?

**Comment by hfiguiere** [comment]
no.

**Comment by hfiguiere** [comment]
bot, build info.beyondallreason.bar

**Comment by tobias47n9e** [comment]
bot, build info.beyondallreason.bar

**Review by hfiguiere — CHANGES_REQUESTED** [review]
Neither appdata nor the .desktop files are installed. Hence the build failing.

File: info.beyondallreason.bar.yml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        url: https://github.com/gajop/spring-launcher
+        commit: 03d42be4796253ad8d61a68cf946e589e1e0deb6
+      - type: patch
+        path: patch/remove-launcher-self-update.patch
+      - type: file
+        path: info.beyondallreason.bar.desktop
+      - type: file
+        url: https://raw.githubusercontent.com/beyond-all-reason/BYAR-Chobby/master/dist_cfg/build/icon.png
+        sha256: 130f61cfa4ca74daa462e46f26279625e7eb3f967ef9bfb62a551b3ea25d0372
+        dest-filename: icon.png
```
Comment (hfiguiere) [inline info.beyondallreason.bar.yml:74]: what you could do instead is just installing the icon in the `bybar-chobby` module.

File: info.beyondallreason.bar.yml [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        node_modules/.bin/electron-builder -- $ELECTRON_BUILDER_ARCH_ARGS --linux --dir --project /run/build/spring-launcher
+      - cp -a dist/linux*unpacked /app/main
+      - install -Dm755 -t /app/bin/ launcher-args.sh
+      - install -Dm755 -t /app/bin/ run.sh
+      - install -Dm644 info.beyondallreason.bar.desktop /app/share/applications/info.beyondallreason.bar.desktop
+      - install -Dm644 icon.png /app/share/icons/hicolor/16x16/apps/info.beyondallreason.bar.png
+      - install -Dm644 icon.png /app/share/icons/hicolor/32x32/apps/info.beyondallreason.bar.png
+      - install -Dm644 icon.png /app/share/icons/hicolor/64x64/apps/info.beyondallreason.bar.png
+      - install -Dm644 icon.png /app/share/icons/hicolor/128x128/apps/info.beyondallreason.bar.png
+      - install -Dm644 icon.png /app/share/icons/hicolor/256x256/apps/info.beyondallreason.bar.png
```
Comment (hfiguiere) [inline info.beyondallreason.bar.yml:62]: what size is the icon? installing the same .png in multiple sizes like that is most likely incorrect. Also you only need 128x128.

File: info.beyondallreason.bar.yml (hunk 4 above)
Comment (hfiguiere) [reply info.beyondallreason.bar.yml:74]: also if the icon change in master, this will break the build.

**Comment by tobias47n9e** [comment]
bot, build info.beyondallreason.bar

File: info.beyondallreason.bar.yml (hunk 5 above)
Comment (tobias47n9e) [reply info.beyondallreason.bar.yml]: Thanks for the info. I will only install 128x128

File: info.beyondallreason.bar.yml (hunk 4 above)
Comment (tobias47n9e) [reply info.beyondallreason.bar.yml]: Good idea. I moved it to byar-chobby stage

File: info.beyondallreason.bar.appdata.xml [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    <metadata_license>CC0-1.0</metadata_license>
+    <project_license>GPL 2.0 and GPL-3.0+</project_license>
+    <description>
+        <p>Beyond All Reason is an RTS</p>
+    </description>
+    <developer_name>BAR Team</developer_name>
+    <url type="homepage">https://www.beyondallreason.info</url>
+    <url type="bugtracker">https://github.com/beyond-all-reason/Beyond-All-Reason/issues</url>
+    <url type="donation">https://www.beyondallreason.info/donate-for-bar</url>
+    <screenshots>
```
[the remark this answers is not in the document]
Comment (tobias47n9e) [reply info.beyondallreason.bar.appdata.xml:15]: I added some

**Comment by tobias47n9e** [comment]
@hfiguiere Do you need the screenshots in an already existing place, or is it ok if I link them to the future repo?

**Comment by tobias47n9e** [comment]
bot, build info.beyondallreason.bar

**Comment by hfiguiere** [comment]
> @hfiguiere Do you need the screenshots in an already existing place, or is it ok if I link them to the future repo?

they have to exist at the time of building.

File: info.beyondallreason.bar.appdata.xml [hunk 7]
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+    <id>info.beyondallreason.bar</id>
+    <name>Beyond All Reason</name>
+    <summary>The Total Annihilation Inspired RTS you've been waiting for</summary>
+    <metadata_license>CC0-1.0</metadata_license>
+    <project_license>GPL 2.0 and GPL-3.0+</project_license>
```
[the remark this answers is not in the document]
Comment (tobias47n9e) [reply info.beyondallreason.bar.appdata.xml:7]: The game says:

The Code: The Spring Engine (which this game runs on) requires that all the code the games use are under the GPL licence. (https://springrts.com)

https://github.com/beyond-all-reason/Beyond-All-Reason/blob/master/LICENSE.md

And Spring Engine says:

see [GNU GPL v2 (gpl-2.0.txt)](gpl-2.0.html)
and [v3 (gpl-3.0.txt)](gpl-3.0.html).

https://github.com/beyond-all-reason/spring/blob/BAR/LICENSE#L16

**Comment by tobias47n9e** [comment]
bot, build info.beyondallreason.bar

**Comment by tobias47n9e** [comment]
bot, build info.beyondallreason.bar

File: info.beyondallreason.bar.appdata.xml (hunk 7 above)
[the remark this answers is not in the document]
Comment (tobias47n9e) [reply info.beyondallreason.bar.appdata.xml]: The devs think that "GPL-2.0-or-later" is the most fitting

**Comment by tobias47n9e** [comment]
I tested the build 58067. When I start it using the console (`flatpak run info.beyondallreason.bar`) I can close the game and can restart it without downloading the game data again. Also I can press "Open install directory" in the launcher and it will open `/home/tobias/.var/app/info.beyondallreason.bar/data`. If I start the game using the Gnome GUI  the "Open install directory" does not work and the data is not persisted. It tries to open `/home/tobias/Beyond All Reason`. That is odd. Have to see why that is happening.

**Comment by tobias47n9e** [comment]
> I tested the build 58067. When I start it using the console (`flatpak run info.beyondallreason.bar`) I can close the game and can restart it without downloading the game data again. Also I can press "Open install directory" in the launcher and it will open `/home/tobias/.var/app/info.beyondallreason.bar/data`. If I start the game using the Gnome GUI the "Open install directory" does not work and the data is not persisted. It tries to open `/home/tobias/Beyond All Reason`. That is odd. Have to see why that is happening.

I solved that. Still had the wrong shell-script in the desktop file.

**Comment by tobias47n9e** [comment]
bot, build info.beyondallreason.bar

**Comment by tobias47n9e** [comment]
@hfiguiere I just tested build 58094 and everything works fine now

**Comment by tobias47n9e** [comment]
Thanks for all your help

**Pull request closed without merging**